src/rust/engine/src/externs/fs.rs

```diff
@@ -426,21 +415,62 @@ impl<'source> FromPyObject<'source> for PyPathGlobs {
   }
 }
 
-#[pyfunction]
-fn match_path_globs(
-  py_path_globs: PyPathGlobs,
-  paths: Vec<String>,
-  py: Python,
-) -> PyResult<Vec<String>> {
-  py.allow_threads(|| {
-    let path_globs = py_path_globs.parse()?;
-    Ok(
-      paths
-        .into_iter()
-        .filter(|p| path_globs.matches(Path::new(p)))
-        .collect(),
-    )
-  })
+// -----------------------------------------------------------------------------
+// FilespecMatcher
+// -----------------------------------------------------------------------------
+
+#[pyclass(name = "FilespecMatcher")]
+#[derive(Debug)]
+pub struct PyFilespecMatcher(FilespecMatcher);
+
+#[pymethods]
+impl PyFilespecMatcher {
+  #[new]
+  fn __new__(includes: Vec<String>, excludes: Vec<String>) -> PyResult<Self> {
+    let matcher = FilespecMatcher::new(includes, excludes).map_err(PyValueError::new_err)?;
+    Ok(Self(matcher))
+  }
+
+  fn __hash__(&self) -> u64 {
+    let mut s = DefaultHasher::new();
+    self.0.include_globs().hash(&mut s);
+    self.0.exclude_globs().hash(&mut s);
+    s.finish()
+  }
+
+  fn __repr__(&self) -> String {
+    let includes = self
+      .0
+      .include_globs()
+      .iter()
+      .map(|pattern| format!("{pattern}"))
```

Comment:
@stuhood is there a better way to call `Display` on the value?

Reply:
`pattern.to_string()` calls `Display`.